src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -477,20 +477,27 @@ private bool CheckLocalValueKind(SyntaxNode node, BoundLocal local, BindValueKin
                     diagnostics.Add(ErrorCode.WRN_AssignmentToLockOrDispose, local.Syntax.Location, localSymbol);
                 }
 
-                if (!localSymbol.IsWritable)
+                // IsWritable means the variable is writable. If this is a ref variable, IsWritable
+                // does not imply anything about the storage location
+                if (localSymbol.RefKind == RefKind.RefReadOnly ||
```

Comment:
I actually always used `IsWritable` on a local as referring to ability to write to the storage. Variable is just a name for some storage. In that sense a writeable variable means that it always represents writeable storage.

I do see the need now for something that represents "variable can be bound to a different storage". It would only make sense for byref locals and arguments. 

I wonder if we can have two helpers with distinct enough names? 
`IsWriteable` is clearly ambiguous now when applied to locals.

Reply:
Basically I am ok with the new `IsWritable`, I just wonder if the name could be a bit more explicit. 
(and whether we need a separate helper for value-writeability)

Reply:
Should I rename it to `IsValueWriteable`?